static/app/components/integrationExternalMappingForm.tsx

```diff
@@ -4,127 +4,153 @@ import capitalize from 'lodash/capitalize';
+        url: dataEndpoint,
         onResults: result => {
-          // For organizations with >100 users, we want to make sure their
+          onResults?.(result, isInline ? mapping?.externalName : mappingKey);
+          // TODO(Leander): The code below only fixes the problem when viewed, not when edited
+          // Pagination still has bugs for results not on initial return of the query
+
+          // For organizations with >100 entries, we want to make sure their
           // saved mapping gets populated in the results if it wouldn't have
           // been in the initial 100 API results, which is why we add it here
-          if (mapping && !result.find(({user}) => user.id === mapping.userId)) {
-            result = [{id: mapping.userId, name: mapping.sentryName}, ...result];
+          if (
+            mapping &&
+            !result.find(entry => {
+              const id = type === 'user' ? entry.user.id : entry.id;
+              return id === mapping[`${type}Id`];
+            })
+          ) {
+            return optionMapper([
+              {id: mapping[`${type}Id`], name: mapping.sentryName},
+              ...sentryNamesMapper(result),
+            ]);
           }
-          this.props.onResults?.(result);
           return optionMapper(sentryNamesMapper(result));
         },
-      });
-    }
-    if (type === 'team') {
-      fields.push({
-        name: 'teamId',
-        type: 'select_async',
+      },
+    ];
+    // We only add the field for externalName if it's the full (not inline) form
+    if (!isInline) {
+      fields.unshift({
+        name: 'externalName',
+        type: 'string',
         required: true,
-        label: tct('Sentry [type]', {type: capitalize(type)}),
-        placeholder: t(`Choose your Sentry Team`),
-        url,
-        onResults: result => {
-          // For organizations with >100 teams, we want to make sure their
-          // saved mapping gets populated in the results if it wouldn't have
-          // been in the initial 100 API results, which is why we add it here
-          if (mapping && !result.find(({id}) => id === mapping.teamId)) {
-            result = [{id: mapping.teamId, name: mapping.sentryName}, ...result];
-          }
-          // The team needs `this.props.onResults` so that we have team slug
-          // when a user submits a team mapping, the endpoint needs the slug
-          // as a path param: /teams/${organization.slug}/${team.slug}/external-teams/
-          this.props.onResults?.(result);
-          return optionMapper(sentryNamesMapper(result));
-        },
+        label: isInline ? undefined : tct('External [type]', {type: capitalize(type)}),
+        placeholder: t(`${type === 'user' ? '@username' : '@org/teamname'}`),
       });
     }
     return fields;
   }
 
-  render() {
-    const {onSubmitSuccess, onCancel, mapping, baseEndpoint, onSubmit} = this.props;
+  get extraFormFieldProps() {
```

Comment:
We provide different props to the fields depending on whether or not it is rendered in line. For non-inline, we just provide a styling prop. For inline, we need to provide some more:

`getData` is used since this form submits onBlur. `saveOnBlur` forms will only submit the edited field, but we need the entire form's data to submit for the API to not 400. `getData` will ensure that the entire model submits when we make the request during `saveOnBlur`

`onBlur` is used to update the model before `saveOnBlur` fires. It's a bit of a hack because of the way models/forms work. `saveOnBlur` (annoyingly) doesn't fire any `onSubmit` or `onPreSubmit` meaning that if we want to run arbitrary code (e.g. updating our endpoint), we have to find another way to run it before firing the request. This was the best method I found.